x-pack/plugins/ml/public/application/jobs/new_job/job_from_lens/create_job.ts

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { IUiSettingsClient, SavedObjectReference } from '@kbn/core/public';
+import type { DataViewsContract } from '@kbn/data-views-plugin/public';
+
+import { Query } from '@kbn/data-plugin/public';
+import { Filter } from '@kbn/es-query';
+
+import type {
+  LensSavedObjectAttributes,
+  FieldBasedIndexPatternColumn,
+  XYDataLayerConfig,
+  IndexPatternPersistedState,
+  GenericIndexPatternColumn,
+  IndexPatternLayer,
+  TermsIndexPatternColumn,
+} from '@kbn/lens-plugin/public';
+
+import type { JobCreatorType } from '../common/job_creator';
+import { createEmptyJob, createEmptyDatafeed } from '../common/job_creator/util/default_configs';
+import { stashJobForCloning } from '../common/job_creator/util/general';
+import { CREATED_BY_LABEL } from '../../../../../common/constants/new_job';
+import { createAbsoluteTimeRange } from '../../../../../common/util/date_utils';
+import { createQueries, getDefaultQuery } from '../utils/new_job_utils';
+import { lensOperationToMlFunction } from './utils';
+
+const COMPATIBLE_SERIES_TYPES = [
```

Comment:
Are the values for these types exported by Lens?

Reply:
https://github.com/elastic/kibana/pull/129762/commits/24518757f81ca73c78332ba4c5e1a5103ad10404